src/main/java/org/influxdb/dto/Point.java

```diff
@@ -269,6 +266,11 @@ void setFields(final Map<String, Object> fields) {
     this.fields = fields;
   }
 
+  public Map<String, Object> getFields() {
```

Comment:
I was hoping you could cite actual challenges / problems with the client to date that require it to be immutable rather than point out an ideal best practice that itself cites trade offs of using such a technique:

> While the cost of creating the builder is unlikely to be noticeable in practice, it could be a problem in some performance-critical situations

I encourage re-examining this practice as the Builder pattern was designed to address issues in Factory implementations, which have changed significantly with the advent of Spring Framework and other DI frameworks.

Ultimately, Point is not immutable (setFields, setPrecision, setTags, setTime, setMeasurement, etc) so I'd think there is room for leeway.